src/CloudFoundry.WinDEA/StagingInstance.cs

```diff
@@ -265,34 +267,226 @@ public void GetBuildpack(StagingStartMessageRequest message, string gitPath, str
+                                continue;
+                            }
+
+                            DEAUtilities.DirectoryCopy(dir, Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+                        }
+
+                        finally
+                        {
+                            buildpackMutex.ReleaseMutex();
+                            buildpackMutex.Dispose();
+                        }
+
+                        Buildpack bp = new Buildpack(Path.Combine(this.Workspace.TempDir, "buildpack"), Path.Combine(this.Workspace.StagedDir, "app"), this.Workspace.Cache, this.Workspace.StagingLogPath);
+                        bool success = bp.Detect(this.Container);
+                        if (success)
+                        {
+                            this.Buildpack = bp;
+                            break;
+                        }
+                        else
+                        {
+                            Directory.Delete(Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+
+                        }
                     }
-                    else
+                }
+
+                if (this.Buildpack == null)
+                {
+                    foreach (string dir in systemBuildpacks)
                     {
-                        Directory.Delete(Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+                        DEAUtilities.DirectoryCopy(dir, Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+                        Buildpack bp = new Buildpack(Path.Combine(this.Workspace.TempDir, "buildpack"), Path.Combine(this.Workspace.StagedDir, "app"), this.Workspace.Cache, this.Workspace.StagingLogPath);
+                        bool success = bp.Detect(this.Container);
+                        if (success)
+                        {
+                            this.Buildpack = bp;
+                            break;
+                        }
+                        else
+                        {
+                            Directory.Delete(Path.Combine(this.Workspace.TempDir, "buildpack"), true);
+                        }
                     }
                 }
 
+
                 if (this.Buildpack == null)
                 {
                     throw new Exception("Unable to detect a supported application type");
                 }
                 Logger.Info("Staging task {0}: Detected buildpack {1}", this.Properties.TaskId, this.Buildpack.Name);
             }
+
             this.Properties.DetectedBuildpack = this.Buildpack.Name;
         }
 
+        public static bool IsWindowsCompatibleBuildpack(string buildpackDir)
+        {
+            var bins = new string[] { "detect", "compile", "release" };
+            string[] pathExt = Environment.GetEnvironmentVariable("PATHEXT").Split(';');
```

Comment:
I'll will add the common windows binaries in a static field.
I skipped .PS1 because it can create problems depending on the system execution policy.
".COM", ".EXE", ".BAT", ".CMD"
Is it ok if I don't include a config filed for this? I think it will just add unnecessary complexity and maybe incompatibilities between different deployments. Even if a buildpack dev wants to call an extra executables, the unsupported binary can be wrapped easily with a .bat.